packages/powersync/lib/src/powersync_database.dart

```diff
@@ -625,6 +642,7 @@ Future<void> _powerSyncDatabaseIsolate(
         uploadCrud: uploadCrud,
         updateStream: updateController.stream,
         retryDelay: args.retryDelay);
+    openedStreamingSync = sync;
```

Comment:
Small question on this. Why not just assign `StreamingSyncImplementation` on line 638 to `openedStreamingSync`?

Reply:
Mostly because it would then need null guards in all other places, since `openedStreamingSync` is nullable.